libs/blocks/global-navigation/utilities/keyboard/index.js

```diff
@@ -25,6 +25,43 @@ const cycleOnOpenSearch = ({ e, isDesktop }) => {
   }
 };
 
+const openProfile = ({ e, el }) => {
+  const button = e.target.closest(`${selectors.signIn}, ${selectors.profileButton}`);
+  if (button?.getAttribute('aria-expanded') === 'false') {
+    e.target.click();
+    el.querySelector(selectors.profileDropdown)?.focus();
+    return true;
+  }
+  return false;
+};
+
+const getProfileItems = ({ e }) => {
+  const focusableElements = document.querySelectorAll(selectors.profileDropdown);
+  if (!focusableElements.length) return { next: -1, prev: -1, items: [] };
+  const items = [...focusableElements];
+  const curr = items.findIndex((element) => element === e.target);
+  return { next: curr + 1, prev: curr - 1, items };
+};
+const focusNextProfileItem = ({ e }) => {
+  const { items, next } = getProfileItems({ e });
+  if (items[next]) items[next].focus();
```

Comment:
It's not so straightforward to test profile-related code from this branch, so I might have some silly questions: I see that `focusPrevProfileItem` closes the profile if a previous element is not found. Should this happen here as well? What happens after tab-ing on the last focusable element of the profile dropdown? Does it close and focus the next navigation item?

Reply:
Hm interestingly enough...
- last element of a profile -> tab -> next item but popup not closed
- last element of a popup (no next main nav item) -> tab -> does nothing

That's wrong in the current (aem/Eurovision milo gnav) version too you shouldn't get stuck using tab. I changed that.

I'll change that to
- last element of a profile -> tab -> close + focus next item
- last element of a popup (no next main nav item) -> tab -> close popup -> tab -> next item (if you wonder why close first and then another tab is needed... well otherwise things get a little messy)

Reply:
Thanks for raising this question, I uncovered some bigger issues when the keyboard navigation isn't loaded or nav items are clicked and the internal state is a bit off.

I refactored a bit of the logic and made some improvements where it made sense such as the flows described above